### ISSUE TITLE
bug: some data samples can't be loaded with yt.load_sample

### ISSUE THREAD

**Issue by neutrinoceros** [yt-project/yt#3178]
Script 
```python
import yt
yt.load_sample("amrvac_blastwave_polar_2D")
```
output 
```
Downloading file 'amrvac_bw_polar_2d.tar.gz' from 'https://yt-project.org/data/amrvac_bw_polar_2d.tar.gz' to '<my data dir>.
100%|███████████████████████████████████████| 468k/468k [00:00<00:00, 1.92GB/s]
Untarring contents of '<my test dir>/amrvac_bw_polar_2d.tar.gz' to '<my test dir>/amrvac_bw_polar_2d.tar.gz.untar'
yt : [INFO     ] 2021-04-09 13:56:59,711 Files located at /<my test di>/amrvac_bw_polar_2d.tar.gz.untar/amrvac/bw_polar_2D0000.dat
yt : [INFO     ] 2021-04-09 13:56:59,711 Default to loading amrvac/bw_polar_2D0000.dat for amrvac_bw_polar_2d dataset
...
FileNotFoundError: No such file or directory: '<my test dir>/amrvac_bw_polar_2d.tar.gz.untar/amrvac_bw_polar_2d/amrvac/bw_polar_2D0000.dat'.
(Also tried '<my test dir>/amrvac_bw_polar_2d.tar.gz.untar/amrvac_bw_polar_2d/amrvac/bw_polar_2D0000.dat').
```

Now with the reimplementation from #3089
```python
import yt
yt.load_sample("amrvac_blastwave_polar_2D")
```
output
```
yt : [INFO     ] 2021-04-09 09:47:44,059 'amrvac_blastwave_polar_2D' is not available locally. Looking up online.
yt : [INFO     ] 2021-04-09 09:47:44,060 Downloading from https://yt-project.org/data/amrvac_bw_polar_2d.tar.gz
100%|███████████████████████████████████████| 468k/468k [00:00<00:00, 1.55GB/s]
yt : [INFO     ] 2021-04-09 09:47:46,092 Untaring downloaded file to '<my data dir>'
...
FileNotFoundError: No such file or directory: '<my data dir>/amrvac_blastwave_polar_2D/amrvac/bw_polar_2D0000.dat'.
```

Note that in both cases the file is properly downloaded but stored in a different location
-  `<my data dir>/ amrvac_bw_polar_2d.tar.gz.untar/amrvac/bw_polar_2D0000.dat`(current implementation)
- `<my data dir>/amrvac/bw_polar_2D0000.dat` (#3089)

I think the real issue here is how amrvac samples are registered (my fault).

_Originally posted by @neutrinoceros in https://github.com/yt-project/yt/issues/3089#issuecomment-816491316_

**Comment by neutrinoceros** [comment]
update: this is also true for the `sedov_tst_0004.h5` data sample.

**Comment by neutrinoceros** [comment]
Not quite fixed in #3280
